pkg/morph/client/multi.go

```diff
@@ -41,6 +43,7 @@ func (x *multiClient) createForAddress(addr string) (*Client, error) {
 	sCli.notary = x.sharedNotary
 
 	c := &Client{
+		cache:        x.sharedCache,
```

Comment:
I think it is a good idea to not share this cache because this way one incorrect response can prevent other clients from working properly. I will create an issue for that.

Reply:
if you mean just creating a new cache for every single client, I have nothing against it and can add it to that PR

just need to add smth with a non-nil cache at that line